### PR TITLE
Fix handling of `settingEditorType`

### PR DESCRIPTION
**Pull request by jtpio**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/13760

## Code changes

Fix handling of `settingEditorType`.

## User-facing changes

**Before**

https://user-images.githubusercontent.com/591645/212176909-ee1e9538-5f0d-470c-8c9e-8895cd3bfacb.mp4

**After**

https://user-images.githubusercontent.com/591645/212176929-ec235d0e-f77e-4966-bfff-e9849b0d769f.mp4

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by jtpio** [comment]
Adding to 3.5.x since it would be nice to backport it for 3.5 and 3.6.

**Comment by jtpio** [comment]
This will help fix the issue spotted downstream in Notebook 7: https://github.com/jupyter/notebook/pull/6316#issuecomment-1380992123

File: packages/settingeditor-extension/src/index.ts [hunk 1]
```diff
@@ -160,13 +160,18 @@ function activate(
       query?: string;
       settingEditorType?: SettingEditorType;
     }) => {
-      void registry.load(plugin.id).then(settings => {
-        args.settingEditorType ??
-        (settings.get('settingEditorType').composite as SettingEditorType) ===
+      if (args.settingEditorType === 'ui') {
+        void commands.execute(CommandIDs.open);
```
Comment (krassowski) [inline packages/settingeditor-extension/src/index.ts:164]:
```suggestion
        void commands.execute(CommandIDs.open, {query: query});
```

**Comment by jtpio** [comment]
I guess the `examples` CI failure is not relevant as it was passing yesterday:

```
[C 2023-01-13 07:44:49.027 ServerApp] Bad config encountered during initialization: /tmp/tmp9jpdh334 is outside root contents directory
```

Probably related to the latest `jupyter_server` release: https://github.com/jupyter-server/jupyter_server/releases/tag/v2.1.0
Which includes https://github.com/jupyter-server/jupyter_server/pull/1173.

File: packages/settingeditor-extension/src/index.ts (hunk 1 above)
Comment (jtpio) [reply packages/settingeditor-extension/src/index.ts]: Thanks!

**Review by fcollonval — APPROVED** [review]
Thanks @jtpio

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.5.x

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.6.x

**Pull request merged**